### DISCUSSION TITLE
NodeJS Query - Include document not returning

### DISCUSSION THREAD

**Discussion by CyberCyclone** [ravendb/ravendb#14342] [Q&A] [answered]
I've been using this as a guide on adding another document to my query result. https://demo.ravendb.net/demos/nodejs/related-documents/query-related-documents

```
const shippedOrders = await session.query({ collection: 'orders' })
    .include('Lines.Product')
    .whereExists('ShippedAt')
    .all();
```

The issue that I'm facing is that it's not including the `'Lines.Product'` document. It's just returning the original field value.

In the studio any included documents aren't swapped out, but shows a different tab with the results. 

Using the RavenDB example, how would I show the included document in the top level document?

Thanks!

**Comment by ppekrol** [comment, marked as answer]
hi @CyberCyclone 

`Lines` is an array. In order to include documents from an array, you need to use `[]`. Try `Lines[].Product` instead in the include. This will include all relevant products in the result of the query. You can load them afterwards using `session.load` method (no round-trip to server will be made).

Hope that this helps!

**Reply by CyberCyclone** [reply]
Thanks for that. I didn't think Load was the way to retrieve them as I hit an error saying I can't use Load more then 30 times in a session. How would I overcome this if I need to do a query on 100 documents, which then needs to Include 1 document each?

I was thinking the included document would be substituted inside the Query documents.

**Reply by ayende** [reply]
The idea here is that `include` doesn't change the shape of objects. Just what is in the session.

You can then utilize the `load` call on the includes values, which aren't counted toward to max requests limits on the session.
